### PR TITLE
chore: make exception for non-cloud artifacts

### PR DESCRIPTION
**Pull request by JoeWang1127**
Fix https://github.com/googleapis/google-cloud-java/issues/9304

### CODE REVIEW

File: docker/owlbot/java/src/fix-poms.py [hunk 1]
```diff
@@ -279,10 +279,14 @@ def update_bom_pom(filename: str, modules: List[module.Module]):
...
+# https://github.com/googleapis/google-cloud-java/issues/9304
 def _proto_group_id(main_artifact_group_id: str) -> str:
-    prefix = "com.google" \
-        if main_artifact_group_id == "com.google.cloud" \
-        else main_artifact_group_id
+    prefix = "com.google"
+    if main_artifact_group_id != "com.google.cloud" and \
+       main_artifact_group_id != "com.google.area120" and \
+       main_artifact_group_id != "com.google.analytics":
+        prefix = main_artifact_group_id
```
Comment (JoeWang1127) [inline docker/owlbot/java/src/fix-poms.py:289]: Python 3.9 doesn't have a `switch` statement.

Comment (suztomo) [reply docker/owlbot/java/src/fix-poms.py:289]: Can you use "in" operator with a list of the 3 elements?

Comment (suztomo) [reply docker/owlbot/java/src/fix-poms.py:289]: Also add source code comment for the reason why this logic is needed.

**Review by suztomo — CHANGES_REQUESTED** [review]
@JoeWang1127 Update the title with "chore:" and ensure you merge with "chore:" to main. This line may appear in the pull request in downstream repositories and without "chore:" it would appear in their release note of irrelevant repositories.

**Comment by JoeWang1127** [comment]
@suztomo changed the PR title.

**Pull request merged**